nixos/modules/services/databases/postgresql.nix

```diff
@@ -862,11 +830,66 @@ in
 
       unitConfig.RequiresMountsFor = "${cfg.dataDir}";
     };
+
+    systemd.services.postgresql-setup = {
+      description = "PostgreSQL Setup Scripts";
+
+      requires = [ "postgresql.service" ];
+      after = [ "postgresql.service" ];
+
+      serviceConfig = {
+        User = "postgres";
+        Group = "postgres";
+        Type = "oneshot";
+        RemainAfterExit = true;
+      };
+
+      path = [ cfg.finalPackage ];
+      environment.PGPORT = builtins.toString cfg.settings.port;
+
+      # Wait for PostgreSQL to be ready to accept connections.
+      script =
+        ''
+          while ! psql -d postgres -c 'DO $$ BEGIN IF pg_is_in_recovery() THEN RAISE read_only_sql_transaction; END IF; END$$;' 2> /dev/null; do
+              if ! systemctl is-active --quiet postgresql.service; then exit 1; fi
```

Comment:
Doesn't this need a `config.systemd.package` in `path`?

I'm wondering if it'd make sense to actually wait for something where we can reasonably test this, even if this means waiting for pgbackrest[1]. I read over this three times before noticing it.

[1] damn it, that means we have a cyclic dependency, right?

Reply:
Ah OK. My reading was that pgbackrest is a way to test the behavior in here properly, but that this fix would also be necessary for the tests to pass. Fine then. Leaving unresolved given the point about test coverage in this PR still stands.